server/src/main/java/org/candlepin/dto/manifest/v1/ProductTranslator.java

```diff
@@ -69,17 +71,37 @@ public ProductDTO populate(ModelTranslator modelTranslator, Product source, Prod
         destination.setAttributes(source.getAttributes());
         destination.setDependentProductIds(source.getDependentProductIds());
 
-        if (modelTranslator != null) {
+        // Translate children products (recursive op)
```

Comment:
That's a very good question -- one we should answer eventually. The manifest API is ancient, and is at the root of a lot of awkward bits of code surrounding import and, to a lesser extent, export.

We should do something where we specify the version in the manifest somehow (manifest file, perhaps), and if it's not present or doesn't match what we expect, we drop down to this current method. That should at least lay the groundwork for removing the current API and supporting code.

Reply:
Using a version variable in the manifest for an if/else flow would mean that we would still have to export all the now-redundant data, because some candlepin-0.9 doing an import still has to do it the old way. The genuine question I have is, is there a limit to how old candlepin imports we need to support? 

If we knew for example, that in 4 years time our manifest api does not need to support importing below candlepin-4.0 or whatever the provided products epic will be released in, then the manifest version makes sense. Otherwise, if we have to support all candlepins no matter how old, there is no point.